tests/model_serving/model_server/keda/test_isvc_keda_scaling.py

```diff
@@ -0,0 +1,201 @@
+            {
+                **BASE_RAW_DEPLOYMENT_CONFIG,
+                "gpu_count": 1,
+                "name": VLLM_MODEL_NAME,
+                "initial_pod_count": INITIAL_POD_COUNT,
+                "final_pod_count": FINAL_POD_COUNT,
+                "metrics_query": VLLM_METRICS_QUERY_REQUESTS,
+                "metrics_threshold": VLLM_METRICS_THRESHOLD_REQUESTS,
+            },
+            id=f"{VLLM_MODEL_NAME}-single-gpu",
+        ),
+    ],
+    indirect=True,
+)
+class TestVllmKedaScaling:
+    def test_vllm_keda_scaling(
+        self,
+        unprivileged_client: DynamicClient,
+        keda_vllm_inference_service: Generator[InferenceService, Any, Any],
+        response_snapshot: Any,
+        prometheus,
+        verify_initial_pod_count,
+        verify_final_pod_count,
+        pod_resource: Pod,
+    ):
+        verify_initial_pod_count(isvc=keda_vllm_inference_service, expected_count=INITIAL_POD_COUNT)
+
+        run_vllm_concurrent_load(
+            pod_name=pod_resource.name,
+            isvc=keda_vllm_inference_service,
+            response_snapshot=response_snapshot,
+            chat_query=CHAT_QUERY,
+            completion_query=COMPLETION_QUERY,
+        )
+
+        verify_final_pod_count(isvc=keda_vllm_inference_service)
+
+        validate_metrics_field(
+            prometheus=prometheus,
+            metrics_query=VLLM_METRICS_QUERY_REQUESTS,
+            expected_value=str(VLLM_METRICS_THRESHOLD_REQUESTS),
+            greater_than=True,
+        )
+
+        verify_keda_scaledobject(
+            client=unprivileged_client,
+            isvc=keda_vllm_inference_service,
+            expected_trigger_type="prometheus",
+            expected_query=VLLM_METRICS_QUERY_REQUESTS,
+            expected_threshold=VLLM_METRICS_THRESHOLD_REQUESTS,
+        )
+
+
+@pytest.mark.parametrize(
+    "unprivileged_model_namespace, ovms_kserve_serving_runtime, ovms_keda_inference_service",
+    [
+        pytest.param(
+            {"name": "ovms-keda"},
+            RunTimeConfigs.ONNX_OPSET13_RUNTIME_CONFIG,
+            {
+                "name": ModelFormat.ONNX,
+                "model-version": ModelVersion.OPSET13,
+                "model-dir": "test-dir",
+                "initial_pod_count": INITIAL_POD_COUNT,
+                "final_pod_count": FINAL_POD_COUNT,
+                "metrics_query": OVMS_METRICS_QUERY,
+                "metrics_threshold": OVMS_METRICS_THRESHOLD,
+            },
+        )
+    ],
+    indirect=True,
+)
+class TestOVMSKedaScaling:
+    def test_ovms_keda_scaling(
+        self,
+        unprivileged_client: DynamicClient,
+        ovms_keda_inference_service: Generator[InferenceService, Any, Any],
+        prometheus,
+        verify_initial_pod_count,
+        verify_final_pod_count,
+    ):
+        verify_initial_pod_count(isvc=ovms_keda_inference_service, expected_count=INITIAL_POD_COUNT)
+
+        run_ovms_concurrent_load(isvc=ovms_keda_inference_service)
```

Comment:
My understanding of the [previous comment](https://github.com/opendatahub-io/opendatahub-tests/pull/327#discussion_r2128989338) from @rnetser is that stressing the workload can be considered as setup. Thus, this `run_ovms_concurrent_load` should be moved to a fixture (e.g named `stressed_ovms_keda_inference_service`).

Then the checks done in `verify_final_pod_count`, `validate_metrics_field` and `verify_keda_scaledobject` can be broken to individual `test_*` functions with all individual funcs requiring such fixture (assuming all 3 need a stressed workload) to fulfill the pattern of one check per test.